### PR TITLE
fix(flipt): return the correct default value when provided in Flipt

### PR DESCRIPTION
**Pull request by erka**
## This PR

Resolve the issue with the returning correct default value

### Related Issues

Fixes flipt-io/flipt/issues/3418

### CODE REVIEW

File: libs/providers/flipt/src/lib/flipt-provider.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ export class FliptProvider implements Provider {
         };
       }
 
-      if (!variantEvaluation.match) {
+      if (!variantEvaluation.match && variantEvaluation.reason != 'DEFAULT_EVALUATION_REASON') {
```
Comment (beeme1mr) [inline libs/providers/flipt/src/lib/flipt-provider.ts:172]: Is it okay if the reason is "TARGETING_MATCH" when using the default value defined in Flipt?

Comment (erka) [reply libs/providers/flipt/src/lib/flipt-provider.ts:172]: @beeme1mr it follows the same pattern as `resolveBooleanEvaluation`.

Also the official OpenFeature documentation says
- `DEFAULT` - no dynamic evaluation occurred or dynamic evaluation yielded no result
- `TARGETING_MATCH` - the result of a dynamic evaluation, such as a rule or specific user-targeting

Is there something I'm missing? I don't see any conflict here.

Comment (beeme1mr) [reply libs/providers/flipt/src/lib/flipt-provider.ts:172]: There's no conflict. I was just confirming that this was the intended behavior.

**Review by beeme1mr — APPROVED** [review]

File: libs/providers/flipt/src/lib/flipt-provider.ts (hunk 1 above)
Comment (erka) [reply libs/providers/flipt/src/lib/flipt-provider.ts:172]: @beeme1mr thank you!

**Pull request merged**